hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/simple/HoodieGlobalSimpleIndex.java

```diff
@@ -91,12 +89,12 @@ private HoodiePairData<String, HoodieRecordGlobalLocation> fetchRecordGlobalLoca
   /**
    * Load all files for all partitions as <Partition, filename> pair data.
    */
-  private List<Pair<String, HoodieBaseFile>> getAllBaseFilesInTable(
-      final HoodieEngineContext context, final HoodieTable hoodieTable) {
+  private HoodieData<Pair<String, HoodieBaseFile>> getAllBaseFilesInTable(
+      final HoodieEngineContext context, final HoodieTable hoodieTable, int parallelism) {
     HoodieTableMetaClient metaClient = hoodieTable.getMetaClient();
     List<String> allPartitionPaths = FSUtils.getAllPartitionPaths(context, config.getMetadataConfig(), metaClient.getBasePath());
     // Obtain the latest data files from all the partitions.
-    return getLatestBaseFilesForAllPartitions(allPartitionPaths, context, hoodieTable);
+    return getLatestBaseFilesForAllPartitions(context.parallelize(allPartitionPaths, parallelism), context, hoodieTable);
```

Comment:
should we use allPartitionPaths.size as the parallelism instead of the configuration parallelism ?

Reply:
```
Math.max(allPartitionPaths.size(), 1));
```

for eg, checkout HoodieIndexUtils L137.